runner/client/src/mill/runner/client/CoursierClient.scala

```diff
@@ -0,0 +1,32 @@
+package mill.runner.client
+import coursier.cache.{ArchiveCache, FileCache}
+import coursier.jvm.{JavaHome, JvmCache, JvmChannel, JvmIndex}
+import coursier.util.Task
+import coursier.Resolve
+
+object CoursierClient {
+  def resolveJavaHome(id: String): java.io.File = {
+    val coursierCache0 = FileCache[Task]()
+    val jvmCache = JvmCache()
+      .withArchiveCache(ArchiveCache().withCache(coursierCache0))
+      .withIndex(jvmIndex0())
+
+    val javaHome = JavaHome().withCache(jvmCache)
+
+    javaHome.get(id).unsafeRun()(coursierCache0.ec)
+  }
+
+  def jvmIndex0(): Task[JvmIndex] = {
+    val coursierCache0 = FileCache[Task]()
+      .withLogger(coursier.cache.loggers.RefreshLogger.create())
```

Comment:
@alexarchambault for some reason this doesn't seem to be showing any progress logger on the terminal. Not sure why